contracts/EverlongStrategy.sol

```diff
@@ -142,17 +142,38 @@ contract EverlongStrategy is BaseStrategy {
     }
 
     /// @dev Attempt to free the '_amount' of 'asset'.
-    /// @dev Any difference between `_amount` and what is actually freed will be
-    ///      counted as a loss and passed on to the withdrawer.
+    ///      - Any difference between `_amount` and what is actually freed will be
+    ///        counted as a loss and passed on to the withdrawer.
+    ///      - Unrealized losses must be calculated and proportionally
+    ///        distributed. Otherwise early withdrawers will incur no losses,
+    ///        and the last to withdraw will incur all losses.
     /// @param _amount The amount of 'asset' to be freed.
     function _freeFunds(uint256 _amount) internal override {
-        // Close all matured positions (if any).
-        uint256 output = _closeMaturedPositions(0);
-
-        // Close immature positions if additional funds need to be freed.
-        if (_amount > output) {
-            _closePositions(_amount - output);
+        // Calculate the current `totalAssets` and retrieve the previous value.
+        uint256 idle = asset.balanceOf(address(this));
+        uint256 currentTotalAssets = calculatePortfolioValue() + idle;
+        uint256 previousTotalAssets = TokenizedStrategy.totalAssets();
```

Comment:
Is this updated when deploying and freeing funds? I'm just wondering if there is any way that this value and the current total assets are measuring different things.

Reply:
It's increased by the deposit amount when a vault adds debt to the strategy, and it's decreased by the amount freed  + any idle liquidity during `free_funds`.

I ~think~ that both of those changes would be reflected similarly for the current and previous `totalAssets`.

On deposit the previous `totalAssets` would be increased by the deposit amount and the current would be increased by the deposit amount less hyperdrive fees.

On `free_funds`, both previous and current `totalAssets` would be decremented by the real amount of assets transferred out of the strategy (NOT the amount requested to be freed afaik) so they should be equally affected.

Seems kind of weird but afaik it's an apples to apples comparison.